modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -74,28 +118,44 @@ def set_module_options(mod)
   end
 
   def is_module_wanted?(mod)
-    (
-      mod.kind_of?(Msf::Exploit::Local) &&
-      mod.has_check? &&
-      is_session_type_compat?(mod) &&
-      is_module_platform?(mod) &&
-      is_module_arch?(mod) &&
-      is_module_options_ready?(mod)
-    )
+    return false unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
+
+    is_session_type_compat?(mod) &&
+    (@validate_platform ? is_module_platform?(mod) : true) &&
+    (@validate_arch ? is_module_arch?(mod) : true) &&
+    is_module_options_ready?(mod)
   end
 
   def setup
     return unless session
 
     print_status "Collecting local exploits for #{session.session_type}..."
 
+    @validate_arch = datastore['ValidateArch']
+    @validate_platform = datastore['ValidatePlatform']
+    @validate_meterpreter_commands = datastore['ValidateMeterpreterCommands']
+
+    @valid_modules_table = Rex::Text::Table.new(
+      'Header' => "\nValid modules for session #{session.sid}:",
```

Comment:
small nit-pick, a table header normally wouldn't contain a new line - you'd normally want to handle that separately 👀